### DISCUSSION TITLE
Cora dataset node order shuffling

### DISCUSSION THREAD

**Discussion by cxw-droid** [pyg-team/pytorch_geometric#8252] [General]
Hi,

I want to shuffle the node order of cora data (edge_index, node label should be changed accordingly). Would you like to recommend a way to do it? It is easy to do node label shuffling accordingly, but changing edge_index seems more difficult. Thanks.

**Comment by EdisonLeeeee** [comment]
```python
perm = torch.randperm(data.num_nodes)
data.x = data.x[perm]
data.y = data.y[perm]
data.edge_index = perm[data.edge_index]
```

**Reply by cxw-droid** [reply]
Thank you. It seems the new graph is different from the original one. The new `edge_index` seems not correct. E.g., if originally node 0, 1 is connected, then it becomes a new edge (perm[0], perm[1]). After permutation, we need to know the new number of original node 0, node 1 and then connect them.

**Reply by EdisonLeeeee** [reply]
The graph is still the same as the associated features and labels for nodes are the same. `perm[0], perm[1]` is just the new number of original 0 and 1 after permutation.

**Comment by cxw-droid** [comment]
> The graph is still the same as the associated features and labels for nodes are the same. `perm[0], perm[1]` is just the new number of original 0 and 1 after permutation.

The original node numbers `perm[0], perm[1]` are changed to the new node numbers `0, 1`, as shown in `data.x = data.x[perm]`, but it seems we cannot say the original `0, 1` are changed to `perm[0], perm[1]`

**Reply by EdisonLeeeee** [reply]
Sorry, you are right. That should work now
```python
perm = torch.randperm(data.num_nodes)
data.x[perm] = data.x.clone()
data.y[perm] = data.y.clone()
data.edge_index = perm[data.edge_index]

```

**Comment by cxw-droid** [comment]
Thank you. Another implementation:
```
perm = torch.randperm(data.num_nodes)
x = data.x[perm]
y = data.y[perm]
new_idx = torch.zeros_like(perm)
new_idx[perm] = torch.arange(data.num_nodes)
edge_index = new_idx[data.edge_index]
```
